main.go

```diff
@@ -401,49 +503,216 @@ func connectAction(ctx *cli.Context) error {
+// and set values in uiSettings, when JSON format is used.
+func setupFormatOption(ctx *cli.Context) error {
+	// This is run after the `app.Before()` has been run,
+	// the uiSettings is already set up for us to modify.
+	format := ctx.String("format")
+	switch format {
+	case "":
+		return nil
+	case "json":
+		uiSettings.isMachineReadable = true
+		uiSettings.isRich = false
+		return nil
+	default:
+		err := fmt.Errorf(
+			"unsupported format: %s (supported formats: %s)",
+			format,
+			`"json"`,
+		)
+		return cli.Exit(err, 1)
+	}
+}
+
+// DisconnectResult is structure holding information about result of
+// disconnect command. The result could be printed in machine-readable format.
+type DisconnectResult struct {
+	Hostname                  string `json:"hostname"`
+	HostnameError             string `json:"hostname_error,omitempty"`
+	UID                       int    `json:"uid"`
+	UIDError                  string `json:"uid_error,omitempty"`
+	RHSMDisconnected          bool   `json:"rhsm_disconnected"`
+	RHSMDisconnectedError     string `json:"rhsm_disconnect_error,omitempty"`
+	InsightsDisconnected      bool   `json:"insights_disconnected"`
+	InsightsDisconnectedError string `json:"insights_disconnected_error,omitempty"`
+	YggdrasilStopped          bool   `json:"yggdrasil_stopped"`
+	YggdrasilStoppedError     string `json:"yggdrasil_stopped_error,omitempty"`
+	exitCode                  int
+	format                    string
+}
+
+// ConnectResult is structure holding information about results
+// of connect command. The result could be printed in machine-readable format.
+type ConnectResult struct {
+	Hostname              string `json:"hostname"`
+	HostnameError         string `json:"hostname_error,omitempty"`
+	UID                   int    `json:"uid"`
+	UIDError              string `json:"uid_error,omitempty"`
+	RHSMConnected         bool   `json:"rhsm_connected"`
+	RHSMConnectError      string `json:"rhsm_connect_error,omitempty"`
+	InsightsConnected     bool   `json:"insights_connected"`
+	InsightsError         string `json:"insights_connect_error,omitempty"`
+	YggdrasilStarted      bool   `json:"yggdrasil_started"`
+	YggdrasilStartedError string `json:"yggdrasil_started_error,omitempty"`
+	exitCode              int
+	format                string
+}
+
+// Error implement error interface for structure DisconnectResult
+func (disconnectResult DisconnectResult) Error() string {
+	var result string
+	switch disconnectResult.format {
+	case "json":
+		data, err := json.MarshalIndent(disconnectResult, "", "    ")
+		if err != nil {
+			return err.Error()
+		}
+		result = string(data)
+	default:
+		result = "error: unsupported document format: " + disconnectResult.format
+	}
+	return result
+}
+
+// printJSONConnectResult tries to print the result of connect as JSON to stdout.
+// When marshaling of systemStatus fails, then error is returned
+func printJSONConnectResult(connectResult *ConnectResult) error {
```

Comment:
Could this function (and likewise for `printJSONDisconnectResult`) be converted
to a `String` method on the type, conforming the type to the `Stringer`
interface? Then values could be passed to `Printf` format strings that include
"%v" and printed when necessary?